src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -1175,20 +1214,54 @@ def validate_setup_script(
             if validation_result["status"] == "FAIL":
                 raise SetupScriptFailedValidation()
 
-    @staticmethod
+    @classmethod
     def get_validation_result(
+        cls,
         console: AbstractConsole,
+        project_root: Path,
+        deploy_root: Path,
+        bundle_root: Path,
+        generated_root: Path,
+        artifacts: list[PathMapping],
         package_name: str,
         package_role: str,
+        package_distribution: str,
+        package_warehouse: str | None,
+        prune: bool,
+        recursive: bool,
+        paths: List[Path] | None,
         stage_fqn: str,
+        post_deploy_hooks: list[PostDeployHook] | None,
+        package_scripts: List[str],
+        policy: PolicyBase,
         use_scratch_stage: bool,
         scratch_stage_fqn: str,
-        deploy_to_scratch_stage_fn: Callable,
     ):
         """Call system$validate_native_app_setup() to validate deployed Native App setup script."""
         if use_scratch_stage:
             stage_fqn = scratch_stage_fqn
-            deploy_to_scratch_stage_fn()
+            cls.deploy(
+                console=console,
+                project_root=project_root,
+                deploy_root=deploy_root,
+                bundle_root=bundle_root,
+                generated_root=generated_root,
+                artifacts=artifacts,
+                bundle_map=None,
```

Comment:
if we require bundle map here, can't we reduce the number parameters?

Reply:
the `deploy()` method can take a `bundle_map` if you have one or call `build_bundle()` to make one. in this case we don't have one, so we'd have to call `build_bundle()` with the same params anyways so it doesn't make a difference